### PR TITLE
`Development`: Add alias for isTestRun for practice participations

### PR DESCRIPTION
**Pull request by JohannesStoehr**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://ls1intum.github.io/Artemis/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/server/).
- [x] I documented the Java code using JavaDoc style.
#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/client/).
- [x] I documented the TypeScript code using JSDoc style.

### Motivation and Context

Since the boolean flag `testRun` in the participation is also determined whether a it is used for practicing in a course exercise, we should make the code more readable by introducing new methods that better reflect this usage in their name.

### Description

This PR adds a new setter and getter for `testRun` but with a different name to reflect the usage for the practice mode.

#### Performance Review
- [ ] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [ ] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/domain/participation/Participation.java [hunk 1]
```diff
@@ -144,6 +144,24 @@ public void setTestRun(boolean testRun) {
         this.testRun = testRun;
     }
 
+    /**
+     * Same as {@link #isTestRun} since {@link Participation#testRun} is used to determine if a participation in a course exercise is used for practice purposes
+     *
+     * @return true if the participation is only used for practicing after the due date
+     */
+    public boolean isPracticeMode() {
```
Comment (Strohgelaender) [inline src/main/java/de/tum/in/www1/artemis/domain/participation/Participation.java:152]:
```suggestion
    @JsonIgnore
    public boolean isPracticeMode() {
```

**Review by Strohgelaender — APPROVED** [review]
code lgtm

File: src/main/webapp/app/entities/participation/student-participation.model.ts [hunk 2]
```diff
@@ -11,3 +11,22 @@ export class StudentParticipation extends Participation {
...
+ * @param studentParticipation the participation to check
+ */
+export function isPracticeMode(studentParticipation: StudentParticipation | undefined): boolean | undefined {
+    return studentParticipation?.testRun;
+}
+
+/**
+ * Stores whether the participation is used for practicing in a course exercise.
+ * @param studentParticipation the participation that should store if it is used for practicing
+ * @param practiceMode true, if it used for practicing
```
Comment (lennart-keller) [inline src/main/webapp/app/entities/participation/student-participation.model.ts:26]:
```suggestion
 * @param practiceMode true, if it is used for practicing
```

**Review by lennart-keller — APPROVED** [review]
Code lgtm, just one typo :)

**Pull request merged**